controller/state.go

```diff
@@ -584,7 +585,10 @@ func (m *appStateManager) CompareAppState(app *v1alpha1.Application, project *v1
 		manifestRevisions = append(manifestRevisions, manifestInfo.Revision)
 	}
 
-	useDiffCache := useDiffCache(noCache, manifestInfos, sources, app, manifestRevisions, m.statusRefreshTimeout, logCtx)
+	serverSideDiff := m.serverSideDiff ||
+		resourceutil.HasAnnotationOption(app, common.AnnotationCompareOptions, "ServerSideDiff=true")
```

Comment:
Do we need a `ServerSideDiff=false` option in case someone wants to opt a particular app out of SSD?

Reply:
That might be useful if someone enables server-side diff at the controller level and wants to disable just for a specific app. Will add. 👍🏻

Reply:
done